packages/appkit/tests/client/public-methods.test.ts

```diff
@@ -584,6 +584,20 @@ describe('Base Public methods', () => {
     expect(setConnectedWalletInfo).toHaveBeenCalledWith(walletInfo, mainnet.chainNamespace)
   })
 
+  it('should set connected wallet info with type', () => {
+    const walletInfo = { name: 'MetaMask', icon: 'icon-url' }
+    const setConnectedWalletInfo = vi.spyOn(AccountController, 'setConnectedWalletInfo')
+    vi.spyOn(ProviderUtil, 'getProviderId').mockReturnValueOnce('WALLET_CONNECT')
+
+    const appKit = new AppKit(mockOptions)
+    appKit.setConnectedWalletInfo(walletInfo, mainnet.chainNamespace)
+
+    expect(setConnectedWalletInfo).toHaveBeenCalledWith(
+      { ...walletInfo, type: 'WALLET_CONNECT' },
+      mainnet.chainNamespace
+    )
+  })
+
```

Comment:
What about email and social provider examples? Let's have test cases for `getEmail()` and `getUsername()` too?